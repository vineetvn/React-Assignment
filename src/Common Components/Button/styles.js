import styled from 'styled-components';


const Button = styled.button `
    line-height: ${props => (props.theme.lineHeight.medium)};
    vertical-align: baseline;
    letter-spacing: 1px;
    margin: 12px 6px 0px 0px;
    font-size: 12px;
    font-weight: 600;
    border: none;
    background-color: ${props => (props.theme.backgroundColor.fbButton)};
    border-top-left-radius: 50px;
    border-bottom-left-radius: 50px;
    border-top-right-radius: 50px;
    border-bottom-right-radius: 50px;
    color: ${props => (props.theme.color.white)};
    &:hover {
        color: ${props => (props.theme.color.darkCharcoal)}; 
        background-color: ${props => (props.theme.backgroundColor.white)};
    }

`

const P = styled.p `
    color: ${props => (props.theme.color.red)};

`
export {Button, P};