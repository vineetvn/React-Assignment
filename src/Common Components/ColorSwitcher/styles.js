import styled from 'styled-components';
import {FiSettings} from 'react-icons/fi';

export const Settings = styled(FiSettings) `
    font-size: ${props => props.theme.fontSize.medium};
    line-height: ${props => props.theme.lineHeight.normal};
    position: absolute;
    top: 0;
    left: 0;
    color: ${props => props.theme.color.white};
    background-color: ${props => props.theme.backgroundColor.black};
    padding: 10px;
`

export const SettingsDiv = styled.div `
    line-height: ${props => props.theme.lineHeight.normal.h2};
    position: fixed;
    background-color: ${props => props.theme.backgroundColor.black};
    width: 200px;
    display: flex;
    flex-wrap: wrap;
    color: ${props => props.theme.color.white};
    top: 20%;
    left: 0;
    z-index: 200;
`

export const H2 = styled.h2 `
    padding-top: 20px;
`

export const Switchers = styled.span `
    background-color: ${props => props.color};
    padding: 3px 10px;
    flex: 12%;
    margin: 1px;
`

export const Div = styled.div `
    padding: 0;
    margin: 2px;
    overflow: hidden;
`