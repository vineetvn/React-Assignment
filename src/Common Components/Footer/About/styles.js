import styled from 'styled-components';

const Div = styled.div`
    flex: 40%;
`
const P = styled.p`
    font-size: ${props => (props.theme.fontSize.small)};
    font-weight: ${props => (props.theme.fontWeight.regular)};
    line-height: ${props => (props.theme.lineHeight.large)};
    color: ${props => (props.theme.color.footerColor)};
    font-family: ${props => (props.theme.fontFamily.Lato)};
`
export { Div, P };