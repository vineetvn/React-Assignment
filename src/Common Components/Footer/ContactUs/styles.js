import styled from 'styled-components';

const Div = styled.div `
    flex: 25%;
    color: ${props => props.theme.color.footerColor};
    font-family: ${props => (props.theme.fontFamily.Lato)};
`
const Div2 = styled.div `
    display: flex;
    flex-direction: column;
    line-height: ${props => (props.theme.lineHeight.medium)};
    font-weight: ${props => (props.theme.fontWeight.regular)};
    font-size: ${props => (props.theme.fontSize.small)};
`
const Span = styled.span `
    color: ${props => (props.theme.color.shadowWhite)};
`
export {Div, Div2, Span};