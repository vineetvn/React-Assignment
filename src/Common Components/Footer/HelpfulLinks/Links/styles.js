import styled from 'styled-components'

const Li = styled.li `
    list-style: none;
    color: ${props => (props.theme.color.footerColor)};
    display: flex;
    align-items: center;
    line-height: ${props => (props.theme.lineHeight.medium)};
`
const A = styled.a `
    text-decoration: none;
    color: ${props => (props.theme.color.footerColor)};
    font-size: ${props => (props.theme.fontSize.small)};
    font-family: ${props => (props.theme.fontFamily.Lato)};
`
export {Li, A};