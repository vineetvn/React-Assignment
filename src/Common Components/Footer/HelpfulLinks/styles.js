import styled from 'styled-components';

const Div = styled.div `
    flex: 35%;
    display: flex;
    flex-direction: column;
`
const Div2 = styled.div `
    display: flex;
    padding-right: 60px;

    @media only screen and (max-width: 450px) {
        flex-direction: column;
    }
`
const Div3 = styled.div `
    flex: 30%;
    display: flex;
    flex-direction: column;
`

export {Div, Div2, Div3};