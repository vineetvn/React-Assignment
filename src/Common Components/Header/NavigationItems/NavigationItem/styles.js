import styled from 'styled-components';
import {Link} from 'react-router-dom';

const Li = styled.li`
    margin: 0;
    box-sizing: border-box;
    display: flex;
    height: 100%;
    align-items: center;
    line-height: 27px;

    @media only screen and (max-width: 900px) {
        display: none;
    }
`

const StyledLink = styled(Link) `
    text-decoration: none;
    height: 100%;
    box-sizing: border-box;
    display: block;
    padding: 26px 10px;
    color: ${props => (props.theme.color.darkCharcoal)};
    font-size: ${props => (props.theme.fontSize.xxxsmall)};
    font-weight: ${props => (props.theme.fontWeight.bold)};
    letter-spacing: ${props => (props.theme.letterSpacing.small)};
    &:hover {
        border-bottom: 2px solid ${props => props.color};
    }
`

export {Li, StyledLink};

