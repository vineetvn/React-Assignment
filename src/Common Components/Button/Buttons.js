import React from 'react'
import { Button, P } from './styles'

export function Buttons(props) {
    let error = null;
    if(props.disabled) {
        error = <P>Please Fill the Form Correctly</P>
    }
    return (
        <>
        <Button type={props.type} disabled= {props.disabled}>{props.children}</Button>
        {error}
        </>
    );
}
