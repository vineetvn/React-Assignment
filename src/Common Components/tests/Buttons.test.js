import React from 'react'

import Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';
import { Button, P } from '../Button/styles';
import {Buttons} from '../Button/Buttons';
import { shallowRender } from '../../testUtils';

configure({adapter: new Adapter()});

describe('Common Component - Button', () => {
    let wrapper;

    // const props = {
    //     type: 'submit',
    //     disabled: false,
    //     children: 'test'
    // }

    beforeEach(() => {
        wrapper = shallowRender(Buttons);
    })

    it('should render the component', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should render a button', () => {
        expect(wrapper.find(Button));
    })

    it('should render error message', () => {
        wrapper.setProps({disabled: true})
        expect(wrapper.find(P));
    })
})