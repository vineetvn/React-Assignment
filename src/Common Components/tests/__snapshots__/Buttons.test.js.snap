// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Component - Button should render the component 1`] = `ShallowWrapper {}`;
